File: Easy/String/1614. Maximum Nesting Depth of the Parentheses.cpp
//1614. Maximum Nesting Depth of the Parentheses ->c++
class Solution {
public:
    int maxDepth(string s) {
        int ans = 0;
        int temp = 0;
        
        for(char i : s){

            if(i == '(') temp++;
            else if(i == ')') temp--;

            ans = max(ans,temp);


        }
        return ans;
        
    }
};

File: Easy/String/1614. Maximum Nesting Depth of the Parentheses.py
#1614. Maximum Nesting Depth of the Parentheses

class Solution:
    def maxDepth(self, s: str) -> int:
        ans = 0
        temp = 0

        for x in s:
            if(x == '(') : temp+=1
            elif(x == ')') : temp-=1
            ans = max(ans,temp)
        
        return ans
